cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -424,11 +425,15 @@ img_mgmt_upload(struct mgmt_ctxt *ctxt)
             return rc;
         }
         img_mgmt_ctxt.len = len;
+        
+        status = IMG_MGMT_STATUS_UPLOAD_START;
     } else {
         if (!img_mgmt_ctxt.uploading) {
             return MGMT_ERR_EINVAL;
         }
 
+        status = IMG_MGMT_STATUS_UPLOAD_ONGOING;
```

Comment:
Did you consider error status? Once it is set as ONGOING, below I can find couple of places where we could leave without a success.

Reply:
I think, I overlooked this one. I will double check it, Thanks!